engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/Patterns.scala

```diff
@@ -79,12 +97,25 @@ object Patterns extends IRPass {
             val consName = consPat.constructor
             val resolution = consName match {
               case qual: IR.Name.Qualified =>
-                val parts = qual.parts.map(_.name)
-                Some(
-                  bindings.resolveQualifiedName(parts)
-                )
+                qual.parts match {
+                  case (_: IR.Name.SelfType) :: (others :+ item) =>
```

Comment:
Do we have a negative test on what if it is not the first in the qualified name?